lib/spack/spack/database.py

```diff
@@ -365,6 +366,32 @@ def _failed_spec_path(self, spec):
         return os.path.join(self._failure_dir,
                             '{0}-{1}'.format(spec.name, spec.full_hash()))
 
+    def clear_all_failures(self):
+        """Force remove all install failure tracking locks and files."""
+        tty.debug('Releasing prefix failure locks')
+        for pkg_id in list(self._prefix_failures.keys()):
+            lock = self._prefix_failures.pop(pkg_id, None)
+            if lock:
+                lock.release_write()
+
+        tty.debug('Removing the prefix failure lock file')
+        try:
+            os.remove(self.prefix_fail_path)
+        except Exception:
+            pass
+
+        tty.debug('Ensure have an (unlocked) prefix failure lock file')
+        fs.touch(self.prefix_fail_path)
+
+        # Remove all failure markings (aka files)
+        tty.debug('Removing prefix failure tracking files')
+        for fail_mark in os.listdir(self._failure_dir):
+            try:
+                os.remove(os.path.join(self._failure_dir, fail_mark))
+            except Exception as exc:
```

Comment:
Use a narrower exception class.  `OSError` at least.

Also, there should probably be a permission check here -- the removing user will need write access to the failure directory to remove the file.  Do we set it up that way?  If this requires a lot of thought we should save the permission check for a future PR.

Reply:
Okay.  Great point about permissions.

Reply:
There's a separate PR for permissions.  See #16614

Reply:
Changed exception to `OSError`.